source/common/filter/config_discovery_impl.h

```diff
@@ -324,14 +339,10 @@ class FilterConfigSubscription
                       const std::string&) override;
   void onConfigUpdateFailed(Config::ConfigUpdateFailureReason reason,
                             const EnvoyException*) override;
+  void updateComplete();
 
   const std::string filter_config_name_;
-  uint64_t last_config_hash_{0ul};
-  ProtobufTypes::MessagePtr last_config_;
-  std::string last_type_url_;
-  std::string last_version_info_;
-  std::string last_factory_name_;
-  SystemTime last_updated_;
+  ConfigVersionConstSharedPtr last_;
```

Comment:
Upon reading the code again, I think this actually doesn't need to be a shared-ptr, but just a struct. It will be similar to the previous fields, but grouped in a struct. It will save a bit of memory, and will cause less confusion (avoiding the need to understand why is this a shared-ptr).

Reply:
I tried it with struct but the issue is that config is captured in the lambda for filter config providers. Capturing by value is not possible because of the `MessagePtr` inside it.